File: 0001-two-sum/0001-two-sum.rs
use std::collections::HashMap;

impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        

        let mut map = HashMap::new();
        let mut result:Vec<i32> = Vec::new();
        for i in 0..nums.len() {
            if (!map.is_empty() && map.contains_key(&nums[i])){
                result.push(i as i32);
                result.push(*map.get(&nums[i]).unwrap() as i32);
                return result;
            }

            map.insert((target - nums[i]), i);

        }

        return result;



    }
}

File: 0001-two-sum/0001-two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """

        my_dict = {}

        for i in range(len(nums)):
            
            compliment = target - nums[i]
            
            if nums[i] in my_dict:
                return [my_dict[nums[i]], i]
            else:
                my_dict[compliment] = i
            

        return [0, 0]

        